query-api/src/main/java/com/redhat/lightblue/query/MapQueryFieldsIterator.java

```diff
@@ -0,0 +1,121 @@
+/*
+ Copyright 2013 Red Hat, Inc. and/or its affiliates.
+
+ This file is part of lightblue.
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.redhat.lightblue.query;
+
+import com.redhat.lightblue.util.Path;
+
+/**
+ * Maps query fields
+ */
+public class MapQueryFieldsIterator extends QueryIterator {
+
+    /**
+     * Override this method to map fields.
+     */
+    protected Path map(Path p) {
+        return p;
+    }
+    
+    @Override
+    protected QueryExpression itrValueComparisonExpression(ValueComparisonExpression q, Path context) {
+        Path p=map(applyContext(context,q.getField()));
+        if(p!=null)
```

Comment:
With the default implementation a new expression is always created.  Could it be updated to only do that when p!=q.field?